packages/edit-site/src/components/page-templates/index.js

```diff
@@ -1,424 +1,10 @@
-					} );
-				}
-			} );
-		}
-
-		// Handle sorting.
-		if ( view.sort ) {
-			filteredTemplates = sortByTextFields( {
-				data: filteredTemplates,
-				view,
-				fields,
-				textFields: [ 'title' ],
-			} );
-		}
-		// Handle pagination.
-		return getPaginationResults( {
-			data: filteredTemplates,
-			view,
-		} );
-	}, [ allTemplates, view, fields ] );
-
-	const resetTemplateAction = useResetTemplateAction();
-	const actions = useMemo(
-		() => [
-			resetTemplateAction,
-			deleteTemplateAction,
-			renameTemplateAction,
-			postRevisionsAction,
-		],
-		[ resetTemplateAction ]
-	);
-
-	const onChangeView = useCallback(
-		( newView ) => {
-			if ( newView.type !== view.type ) {
-				newView = {
-					...newView,
-					layout: {
-						...defaultConfigPerViewType[ newView.type ],
-					},
-				};
-			}
+import { TEMPLATE_POST_TYPE } from '../../utils/constants';
 
-			setView( newView );
-		},
-		[ view.type, setView ]
-	);
+import DataviewsTemplatesTemplateParts from './dataviews-templates-template-parts';
 
-	return (
-		<>
-			<Page
-				className={
-					view.type === LAYOUT_LIST
-						? 'edit-site-template-pages-list-view'
-						: null
-				}
-				title={ __( 'Templates' ) }
-				actions={
-					<AddNewTemplate
-						templateType={ TEMPLATE_POST_TYPE }
-						showIcon={ false }
-						toggleProps={ { variant: 'primary' } }
-					/>
-				}
-			>
-				<DataViews
-					paginationInfo={ paginationInfo }
-					fields={ fields }
-					actions={ actions }
-					data={ data }
-					isLoading={ isLoadingData }
-					view={ view }
-					onChangeView={ onChangeView }
-					onSelectionChange={ onSelectionChange }
-					deferredRendering={
-						! view.hiddenFields?.includes( 'preview' )
-					}
-				/>
-			</Page>
-			{ view.type === LAYOUT_LIST && (
-				<Page>
-					<div
-						className="edit-site-template-pages-preview"
-						tabIndex={ 0 }
-						role="button"
-						onKeyDown={ ( event ) => {
-							const { keyCode } = event;
-							if ( keyCode === ENTER || keyCode === SPACE ) {
-								history.push( {
-									postId: templateId,
-									postType: TEMPLATE_POST_TYPE,
-									canvas: 'edit',
-								} );
-							}
-						} }
-						onClick={ () =>
-							history.push( {
-								postId: templateId,
-								postType: TEMPLATE_POST_TYPE,
-								canvas: 'edit',
-							} )
-						}
-					>
-						{ templateId !== null ? (
-							<PostPreview
-								postId={ templateId }
-								postType={ TEMPLATE_POST_TYPE }
-							/>
-						) : (
-							<div
-								style={ {
-									display: 'flex',
-									flexDirection: 'column',
-									justifyContent: 'center',
-									textAlign: 'center',
-									height: '100%',
-								} }
-							>
-								<p>{ __( 'Select a template to preview' ) }</p>
-							</div>
-						) }
-					</div>
-				</Page>
-			) }
-		</>
-	);
+export default function PageTemplates() {
+	return <DataviewsTemplatesTemplateParts postType={ TEMPLATE_POST_TYPE } />;
 }
```

Comment:
Is this file still necessary, I think if you remove it you may end up with a better "git diff". Also maybe you should rename the folder since it now contains the reusable component.